### PR TITLE
feat:replaced setup.py with pyproject.toml

### PR DESCRIPTION
**Pull request by alirashidAR**
# Pull Request

## Description

Worked on #9 

Migrated the project from setup.py to pyproject.toml to comply with modern Python packaging standards (PEP 518). This change simplifies the configuration of the package and is more maintainable for future development.

## How Has This Been Tested?

- Created a virtual environment and installed the package using pip install .
 
- Installed optional development dependencies with pip install .[dev]

- Ran the existing test suite, including tests/test_fetch_data.py, to ensure everything was set up correctly and no issues occurred with the installation or dependencies.

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[build-system]
+requires = ["setuptools>=61.0", "wheel"]
```
Comment (siddharth7113) [inline pyproject.toml:2]: Great work on this PR! 
I noticed that `wheel` is included as a build dependency. While it’s definitely helpful for distribution purposes, I was wondering if it’s something we need at this stage of the project.
 @peterdudfield, could you share your thoughts on whether `wheel` is necessary for now or if it’s more of a forward-looking addition?

**Review by siddharth7113 — APPROVED** [review]
Everything looks good to me, and the migration to `pyproject.toml` is well-implemented. I left a small comment regarding the inclusion of `wheel` as a build dependency—just seeking clarification if it's needed at this stage or more forward-looking. This isn’t blocking, so approving this PR.

File: pyproject.toml (hunk 1 above)
Comment (peterdudfield) [reply pyproject.toml:2]: Yea, I'd probably leave wheel out for the moment

Comment (alirashidAR) [reply pyproject.toml:2]: Alright thank you for your input @peterdudfield . @siddharth7113 work on top of this PR or should i make the necessay changes .

Comment (alirashidAR) [reply pyproject.toml:2]: @siddharth7113 sure !

Comment (alirashidAR) [reply pyproject.toml]: @peterdudfield , removed the build dependency from the pyproject.toml file

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Would you like to merge @alirashidAR ?

**Comment by alirashidAR** [comment]
> Would you like to merge @alirashidAR ?

Yes , thank you for the guidance !

**Comment by peterdudfield** [comment]
@all-contributors please add @siddharth7113  for review

**Pull request merged**